### PR TITLE
enable distribution via pip/setuptools

### PR DESCRIPTION
**Pull request by Mic92**

### CODE REVIEW

**Comment by Mic92** [comment]
> Hi Jörg,
> Thanks a lot for your work.
> I'm not familiar with setuptools, but it looks to me that integrating that is a good idea.
> There are a few aspects that I would like to fix before merging.
> 
>     1. Can we avoid creating the `kconfig_hardened_check` directory? I would rather have `bin` and `config_files`.
> 

No one needs a distinct module to put the python code in to avoid conflicts with other installed python packages.

>     2. What is the purpose of splitting the code onto `bin/kconfig_hardened_check` and `kconfig_hardened_check/__init__.py`? Is it some special python feng-shui? (I'm asking because I'm just a kernel developer)
> 

`bin/kconfig_hardened_check` is for people just checking out the repository and running the script without installing it. If you install it with `setuptools`,
it will generate its own wrapper that will eventually load `kconfig_hardened_check/__init__.py`.

>     3. I would like to split setuptools integration and the code refactoring onto separate commits. Moreover, I don't understand the `List[Any]` changes.

`List[Any]` is a type annotation. When you use a typechecker like mypy you can typecheck your code that way.

> 
>     4. Are you sure that the classifiers in `setup.cfg` are correct? It looks like some of them don't fit this project.


> 
>     5. The `package_data` in `setup.cfg` misses some files in the repository. Is it ok?
> 
> 
> Thanks!

File: setup.cfg [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+description = A script for checking the hardening options in the Linux kernel config
+long-description = file: README.md
+license = GPL-3.0-or-later
+license-file = LICENSE
+platform = any
+classifiers =
+    Development Status :: 5 - Production/Stable
+    Environment :: Console
+    Topic :: Utilities
+    License :: OSI Approved :: MIT License
```
Comment (Mic92) [inline setup.cfg]: This one should be dropped.

File: setup.cfg [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    Operating System :: OS Independent
```
Comment (Mic92) [inline setup.cfg]: This one should be changed to Linux. Anything else that should be changed?

**Comment by a13xp0p0v** [comment]
Hello @Mic92,
I carefully reimplemented your proof-of-concept in a set of separate commits.
Fixed mistakes in setup.cfg, added MANIFEST.in, fixed issues with global variables.
Thank you very much, I learned a lot!

**Pull request closed without merging**